packages/compass-saved-aggregations-queries/src/components/open-item-modal.tsx

```diff
@@ -0,0 +1,198 @@
+      disabled={isLoading}
+      usePortal={false}
+      className={select}
+    >
+      {items.map((item) => (
+        <Option key={item} value={item}>
+          {item}
+        </Option>
+      ))}
+    </Select>
+  );
+};
+
+const mapDatabaseState: MapStateToProps<
+  Pick<AsyncItemsSelectProps, 'items' | 'selectedItem' | 'isLoading'>,
+  Pick<AsyncItemsSelectProps, 'name' | 'label'>,
+  RootState
+> = ({ openItem: { databases, databasesStatus, selectedDatabase } }) => {
+  return {
+    items: databases,
+    selectedItem: selectedDatabase,
+    isLoading: ['initial', 'fetching'].includes(databasesStatus),
+  };
+};
+
+const DatabaseSelect = connect(mapDatabaseState, { onChange: selectDatabase })(
+  AsyncItemsSelect
+);
+
+const mapCollectionState: MapStateToProps<
+  Pick<AsyncItemsSelectProps, 'items' | 'selectedItem' | 'isLoading'>,
+  Pick<AsyncItemsSelectProps, 'name' | 'label'>,
+  RootState
+> = ({ openItem: { collections, collectionsStatus, selectedCollection } }) => {
+  return {
+    items: collections,
+    selectedItem: selectedCollection,
+    isLoading: ['initial', 'fetching'].includes(collectionsStatus),
+  };
+};
+
+const CollectionSelect = connect(mapCollectionState, {
+  onChange: selectCollection,
+})(AsyncItemsSelect);
+
+type OpenItemModalProps = {
+  namespace: string;
+  itemType: string;
+  itemName: string;
+  isModalOpen: boolean;
+  isSubmitDisabled: boolean;
+  onSubmit(): void;
+  onClose(): void;
+};
+
+const modalContent = css({
+  display: 'grid',
+  gridTemplateAreas: `
+    'description description'
+    'database collection'
+  `,
+  gridAutoColumns: '1fr',
+  rowGap: spacing[4],
+  columnGap: spacing[3],
+});
+
+const description = css({
+  gridArea: 'description',
+});
+
+const databaseSelect = css({
+  gridArea: 'database',
+});
+
+const collectionSelect = css({
+  gridArea: 'collection',
+});
+
+const OpenItemModal: React.FunctionComponent<OpenItemModalProps> = ({
+  namespace,
+  itemType,
+  itemName,
+  isModalOpen,
+  isSubmitDisabled,
+  onClose,
+  onSubmit,
+}) => {
+  return (
+    <ConfirmationModal
+      open={isModalOpen}
+      onCancel={onClose}
+      onConfirm={onSubmit}
+      title="Select Namespace"
+      buttonText="Open"
+      submitDisabled={isSubmitDisabled}
+    >
+      <div className={modalContent}>
+        <div className={description}>
+          Namespace <strong>{namespace}</strong> for the saved {itemType}{' '}
+          <strong>{itemName}</strong> doesn&rsquo;t exist in&nbsp;the current
+          cluster. Please select another namespace to&nbsp;open saved {itemType}
+          .
+        </div>
+        <div className={databaseSelect}>
+          <DatabaseSelect name="database" label="Database"></DatabaseSelect>
+        </div>
+        <div className={collectionSelect}>
+          <CollectionSelect
+            name="collection"
+            label="Collection"
+          ></CollectionSelect>
+        </div>
+      </div>
+    </ConfirmationModal>
```

Comment:
@mmarcon @Sgrinfy does this copy looks good to you?

Reply:
![image](https://user-images.githubusercontent.com/761042/151932495-52ebb23a-3885-4b15-9b47-a98cc8a7fb49.png)

Here's a static screen to make it easier to review.